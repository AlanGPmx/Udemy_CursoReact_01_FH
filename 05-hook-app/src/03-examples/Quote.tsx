import {useLayoutEffect, useRef, MutableRefObject, useState} from 'react';

export const Quote = (props: any) => {
	const {quote, author} = props;

	const pRef = useRef() as MutableRefObject<HTMLInputElement>;

	const [boxSize, setBoxSize] = useState({width: 0, height: 0});

	useLayoutEffect(() => {
		const {width, height} = pRef.current.getBoundingClientRect();

		setBoxSize({width, height});
	}, []);

	return (
		<>
			<blockquote className='blockquote row' style={{display: 'flex'}}>
				<p ref={pRef} className='mb-2 col-auto'>
					{quote}
				</p>
				<footer className='blockquote-footer col-12'>{author}</footer>
			</blockquote>

			<hr />

			<code>{JSON.stringify(boxSize)}</code>
		</>
	);
};
