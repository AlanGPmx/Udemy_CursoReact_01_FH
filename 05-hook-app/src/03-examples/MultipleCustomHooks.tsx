import {useFetch, useCounter} from '../hooks';
import {LoadingQuote} from './LoadingQuote';
import {Quote} from './Quote';

export const MultipleCustomHooks = () => {
	const {counter, incrementCounter} = useCounter(1);

	const {data, isLoading} = useFetch(
		'https://www.breakingbadapi.com/api/quotes/' + counter
	);

	const {author = '', quote = ''} = data ? data[0] : {};

	return (
		<>
			<h2 className='text-center mb-5'>5.- Breaking Bad Quotes</h2>

			{isLoading ? <LoadingQuote /> : <Quote author={author} quote={quote} />}

			<button
				className='btn btn-primary float-end'
				disabled={isLoading}
				onClick={() => incrementCounter()}>
				Siguiente frase...
			</button>
		</>
	);
};
