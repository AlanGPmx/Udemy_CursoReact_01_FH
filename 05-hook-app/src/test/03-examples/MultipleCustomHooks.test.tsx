import {render, screen} from '@testing-library/react';
import {MultipleCustomHooks} from '../../03-examples/MultipleCustomHooks';

describe('Pruebas en <MultipleCustomHooks />', () => {
	test('Debe mostrar el componente por defecto sin snapshot', () => {
		render(<MultipleCustomHooks />);
		expect(screen.getByText('Cargando...')).toBeInTheDocument();
		expect(screen.getByText('5.- Breaking Bad Quotes')).toBeInTheDocument();
		const boton = screen.getByRole('button', {
			name: 'Siguiente frase...',
		}) as HTMLInputElement;
		expect(boton.disabled).toBeTruthy();
	});

	test('Debe mostrar el componente por defecto con snapshot', () => {
		const {container} = render(<MultipleCustomHooks />);
		expect(container).toMatchSnapshot();
	});
});
