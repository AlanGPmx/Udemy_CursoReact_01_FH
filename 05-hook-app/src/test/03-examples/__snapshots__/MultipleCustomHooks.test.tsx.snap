// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHooks /> Debe mostrar el componente por defecto con snapshot 1`] = `
<div>
  <h2
    class="text-center mb-5"
  >
    5.- Breaking Bad Quotes
  </h2>
  <div
    class="alert alert-info text-center"
  >
    Cargando...
  </div>
  <button
    class="btn btn-primary float-end"
    disabled=""
  >
    Siguiente frase...
  </button>
</div>
`;
